### PR TITLE
Added some missing color picker settings

### PR DESCRIPTION
**Pull request by tmcconechy**
Added some missing color picker options noted on https://jira.infor.com/browse/SOHO-8007

### CODE REVIEW

File: src/soho/colorpicker/soho-colorpicker.component.ts [hunk 1]
```diff
@@ -88,6 +96,70 @@ export class SohoColorPickerComponent extends BaseControlValueAccessor<any> impl
...
+  @Input() set uppercase(value: boolean) {
+    this.isUppercase = value;
+    this.options.uppercase = value;
+
+    if (this.colorpicker) {
+      this.colorpicker.settings.uppercase = value;
+    }
+
+    if (this.colorpicker) {
+      this.colorpicker.updated(this.options);
```
Comment (pwpatton) [inline src/soho/colorpicker/soho-colorpicker.component.ts]: would like to see a `callUpdatedAfterCheck: boolean` flag then call the updated() function in a runOutsideAngular in ngAfterViewChecked.

Example
In the @Inputs just set the flag.
```
  if (this.colorpicker) {
      this.callUpdatedAfterCheck = true;
  }
```

In ngAfterViewChecked check the flag and then call updated and reset the flag
```
ngAfterViewChecked() {
  if (this.callUpdatedAfterCheck) {
    this.ngZone.runOutsideAngular(() => this.colorpicker.updated(this.options));
    this.callUpdatedAfterCheck = false;
  }
}
```

Maybe not super important right now but it's a good way to ensure that updated() is not called for each input setting but instead at the end of a tick() after all inputs have been set.

Comment (tmcconechy) [reply src/soho/colorpicker/soho-colorpicker.component.ts]: I can make the change. Is there a reference component done with all ngZone updates and the change to only trigger on ngAfterViewChecked ?

Comment (bthharper) [reply src/soho/colorpicker/soho-colorpicker.component.ts]: I would take a look at dropdown, as this has both the run updated on check boolean, and the NgZone code.

**Comment by bthharper** [comment]
Looking at the `SohoColorPickerComponentValueAccessorDelegator `, which for obvious reasons mimics the dropdown implementation, I wonder if there is a case for creating a generic delegate used by both.  This looks like a repeating pattern.

File: src/soho/colorpicker/soho-colorpicker.component.ts [hunk 2]
```diff
@@ -210,4 +342,74 @@ export class SohoColorPickerComponent extends BaseControlValueAccessor<any> impl
...
+/**
+ * Provides a 'wrapper' around the {ControlValueAccessor} added by
+ * angular when handling `select` elements.
+ *
+ * This class allows the {SohoColorPickerComponent} to interoperate with
+ * the {ControlValueAccessor}.  Specifically, providing access to the
+ * onChange function, which we must call when the value of the colorpicker
+ * is modified.
+ *
+ * It also exposes the encoding used storing complex objects as
```
Comment (bthharper) [inline src/soho/colorpicker/soho-colorpicker.component.ts]: Delete this line (and the next)?

Comment (pwpatton) [reply src/soho/colorpicker/soho-colorpicker.component.ts]: not sure, why these lines?

File: src/soho/colorpicker/soho-colorpicker.component.ts [hunk 3]
```diff
@@ -79,15 +104,67 @@ export class SohoColorPickerComponent extends BaseControlValueAccessor<any> impl
...
       } else {
-        this.colorpicker.readonly();
+        this.ngZone.runOutsideAngular(() => this.colorpicker.readonly());
         this.isEditable = false;
       }
     }
   }
 
+  /**
+   * Enables or disables forceing upper case hex.
```
Comment (fitzorama) [inline src/soho/colorpicker/soho-colorpicker.component.ts]: forcing

**Comment by tmcconechy** [comment]
Fixed spelling please re-approve.

**Pull request merged**